instaclustr/resource_cluster.go

```diff
@@ -383,6 +403,35 @@ func resourceCluster() *schema.Resource {
 	}
 }
 
+func getNodeSize(d *schema.ResourceData, bundles []Bundle) (string, error) {
+	for i, bundle := range bundles {
+		if bundle.Bundle == "ELASTICSEARCH" {
+			if len(bundle.Options.MasterNodeSize) == 0 {
+				return "", fmt.Errorf("[ERROR] 'master_node_size' is required in the bundle option.")
+			}
+			dedicatedMaster := bundle.Options.DedicatedMasterNodes != nil && *bundle.Options.DedicatedMasterNodes
+			if dedicatedMaster {
+				if len(bundle.Options.DataNodeSize) == 0 {
+					return "", fmt.Errorf("[ERROR] Elasticsearch dedicated master is enabled, 'data_node_size' is required in the bundle option.")
+				}
+				return bundle.Options.DataNodeSize, nil
+			} else {
+				if len(bundle.Options.DataNodeSize) != 0 && bundle.Options.DataNodeSize != bundle.Options.MasterNodeSize {
+					return "", fmt.Errorf("[ERROR] When 'dedicated_master_nodes' is not true , data_node_size can be either null or equal to master_node_size.")
+				}
+				size := bundle.Options.MasterNodeSize
+				bundles[i].Options.MasterNodeSize = ""
```

Comment:
This method is only being used when creating clusters. I have run the newly added acceptance tests locally and they all passed. But I didn't run the old acc test because I didn't have the correct config.